common/v2/features/ReceiveAssets/ReceiveAssets.tsx

```diff
@@ -0,0 +1,255 @@
+                  <AccountDropdown
+                    name={field.name}
+                    value={field.value}
+                    accounts={accounts}
+                    onSelect={(option: IExtendedAccount) => {
+                      form.setFieldValue(field.name, option);
+                      setRequestAddress(option.address);
+                    }}
+                  />
+                )}
+              />
+            </Fieldset>
+            <AssetFields>
+              <Amount>
+                <SLabel htmlFor="amount">Amount</SLabel>
+                <Field
+                  name="amount"
+                  validate={validateAmount}
+                  render={({ field, form }: FieldProps<typeof initialValues>) => (
+                    <FullWidthInput
+                      value={field.value}
+                      onChange={({ target: { value } }) => form.setFieldValue(field.name, value)}
+                      placeholder="0.00"
+                    />
+                  )}
+                />
+              </Amount>
+              <Asset>
+                <SLabel htmlFor="asset">Asset</SLabel>
+                <Field
+                  name="asset"
+                  render={({ field, form }: FieldProps<typeof initialValues>) => (
+                    <StyledSelect
+                      name="Assets"
+                      className="select-container"
+                      options={assetOptions}
+                      value={field.value}
+                      onChange={(option: Option) => {
+                        form.setFieldValue(field.name, option);
+                        if (option.label) {
+                          setAssetName(option.label);
+                        }
+                      }}
+                    />
+                  )}
+                />
+              </Asset>
+            </AssetFields>
+            {errors.amount && (
+              <ErrorMessage>{' ' + translateRaw('RECEIVE_FORM_ERROR')}</ErrorMessage>
+            )}
+            {parseFloat(amount) >= 0 &&
+              selectedAsset &&
+              requestAddress &&
+              network && (
+                <>
+                  <Divider />
+                  <Fieldset>
+                    <SLabel>Payment Code</SLabel>
+                    <FieldsetBox>
+                      <Copyable
+                        text={
+                          isAssetToken(selectedAsset.type) &&
+                          selectedAsset.contractAddress &&
+                          selectedAsset.decimal
+                            ? buildEIP681TokenRequest(
+                                requestAddress,
+                                selectedAsset.contractAddress,
+                                network.chainId,
+                                amount,
+                                selectedAsset.decimal
+                              )
+                            : buildEIP681EtherRequest(requestAddress, chainId, amount)
```

Comment:
Component doesn't need to be aware of this logic
Create a method in `StoreProvider` that takes `account, asset, amount` and returns a `txObject`.
That would be the best place to handle the business logic aka. knowing if it's a base asset or a token